src/Wallabag/UserBundle/Controller/ManageController.php

```diff
@@ -62,14 +62,29 @@ public function newAction(Request $request)
      */
     public function editAction(Request $request, User $user)
     {
+        $userManager = $this->container->get('fos_user.user_manager');
+
         $deleteForm = $this->createDeleteForm($user);
-        $editForm = $this->createForm('Wallabag\UserBundle\Form\UserType', $user);
-        $editForm->handleRequest($request);
+        $form = $this->createForm('Wallabag\UserBundle\Form\UserType', $user);
+        $form->handleRequest($request);
 
-        if ($editForm->isSubmitted() && $editForm->isValid()) {
-            $em = $this->getDoctrine()->getManager();
-            $em->persist($user);
-            $em->flush();
+        // `googleTwoFactor` isn't a field within the User entity, we need to define it's value in a different way
+        if ($this->getParameter('twofactor_auth') && true === $user->isGoogleAuthenticatorEnabled() && false === $form->isSubmitted()) {
+            $form->get('googleTwoFactor')->setData(true);
+        }
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            // handle creation / reset of the OTP secret if checkbox changed from the previous state
+            if ($this->getParameter('twofactor_auth')) {
+                if (true === $form->get('googleTwoFactor')->getData() && false === $user->isGoogleAuthenticatorEnabled()) {
+                    $user->setGoogleAuthenticatorSecret($this->get('scheb_two_factor.security.google_authenticator')->generateSecret());
+                    $user->setEmailTwoFactor(false);
```

Comment:
Would it be an option to allow setting of both? Or is there a specific reason why OTP takes preference over email?

I've seen other 2FA implementations that allow OTP and U2F for example, to be able to decide which one to use.

Reply:
Should we allow more than on OTP per account? I don't know. I decided that one take on the other one.
As far as I remember I usually choose only one OTP solution when many are available.

I'll also need to check if both can be enabled.

Reply:
For software-based 2FA it might not be necessary, but I've been super grateful for a second option when using U2F and OTP, as some browsers aren't fully U2F compatible yet. So having only U2F enabled might lock you out in some cases when not using the environment where it was set up.

It was more a question of understanding why only 1.
I think it makes sense to leave it as is and if the need should arise in the future by many users, to then have another look :+1: